src/cosmospy/_transaction.py

```diff
@@ -44,64 +49,86 @@ def __init__(
 
     def get_pushable(self) -> str:
-        pubkey = privkey_to_pubkey(self._privkey)
-        base64_pubkey = base64.b64encode(pubkey).decode("utf-8")
-        pushable_tx = {
-            "tx": {
-                "msg": self._msgs,
-                "fee": {
-                    "gas": str(self._gas),
-                    "amount": [{"denom": self._fee_denom, "amount": str(self._fee)}],
-                },
-                "memo": self._memo,
-                "signatures": [
-                    {
-                        "signature": self._sign(),
-                        "pub_key": {"type": "tendermint/PubKeySecp256k1", "value": base64_pubkey},
-                        "account_number": str(self._account_num),
-                        "sequence": str(self._sequence),
-                    }
-                ],
-            },
-            "mode": self._sync_mode,
-        }
-        return json.dumps(pushable_tx, separators=(",", ":"))
-
-    def _sign(self) -> str:
-        message_str = json.dumps(self._get_sign_message(), separators=(",", ":"), sort_keys=True)
-        message_bytes = message_str.encode("utf-8")
+        self._tx_raw.body_bytes = self._tx_body.SerializeToString()
+        self._tx_raw.auth_info_bytes = self._get_auth_info().SerializeToString()
+        self._tx_raw.signatures.append(self._get_signatures())
+        raw_tx = self._tx_raw.SerializeToString()
+        tx_bytes = bytes(raw_tx)
+        tx_b64 = base64.b64encode(tx_bytes).decode("utf-8")
+        return json.dumps(
+            {"jsonrpc": "2.0", "id": 1, "method": self._sync_mode, "params": {"tx": tx_b64}}
+        )
 
+    def _get_signatures(self):
         privkey = ecdsa.SigningKey.from_string(self._privkey, curve=ecdsa.SECP256k1)
         signature_compact = privkey.sign_deterministic(
-            message_bytes, hashfunc=hashlib.sha256, sigencode=ecdsa.util.sigencode_string_canonize
+            self._get_sign_doc().SerializeToString(),
+            hashfunc=hashlib.sha256,
+            sigencode=ecdsa.util.sigencode_string_canonize,
         )
+        return signature_compact
+
+    def _get_sign_doc(self):
+        sign_doc = tx.SignDoc()
+        sign_doc.body_bytes = self._tx_body.SerializeToString()
+        sign_doc.auth_info_bytes = self._get_auth_info().SerializeToString()
+        sign_doc.chain_id = self._chain_id
+        sign_doc.account_number = self._account_num
+        return sign_doc
+
+    def _get_auth_info(self):
+        _auth_info = tx.AuthInfo()
+        _auth_info.signer_infos.append(self._get_signer_infos(self._get_pubkey()))
+        _auth_info.fee.gas_limit = self._gas
+        _auth_info.fee.amount.append(self._get_fee())
+        return _auth_info
+
+    def _get_fee(self):
+        _fee = coin.Coin()
+        _fee.amount = str(self._fee)
+        _fee.denom = self._fee_denom
+        return _fee
+
+    def _get_pubkey(self):
+        pubkey_bytes = privkey_to_pubkey(self._privkey)
+        _pubkey = pubkey.PubKey()
```

Comment:
There shouldn't be a reason to use private variables inside the function scope (applies to other methods too)